neurolib/optimize/evolution/evolution.py

```diff
@@ -678,8 +777,11 @@ def saveEvolution(self, fname=None):
         :type fname: str, optional
         """
         import dill
-        fname = fname or os.path.join("data/", "evolution-" + self.trajectoryName + ".dill")
-        dill.dump(self, open(fname, "wb"))        
+
+        fname = fname or os.path.join(
+            "data/", "evolution-" + self.trajectoryName + ".dill"
+        )
+        dill.dump(self, open(fname, "wb"))
```

Comment:
are you trying to dump whole class? maybe try `self.__dict__`... each class have `__dict__` method which is a dictionary of all attributes... when I am saving whole class from delfi, I save it to pickle (that shouldn't matter whether `pickle` or `dill`) with `dump(self.__dict__, fname)`

Reply:
pickle works only in serializable data.... I forgot which but one of the many objects (pypet?) didn't quite like to be pickled unfortunately.

Reply:
yes, you're right, might be something is not pickable, then dill is indeed an option ;)
still, I'd try dilling `self.__dict__`, maybe that'll work

Reply:
Referring to `self.__dict__`, how would you load the original instance back into memory and replace it with `self`?

Reply:
Guess I'm gonna leave it as is for now, since it seems to work!